hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/test/java/org/hypertrace/traceenricher/enrichment/enrichers/BackendEntityEnricherTest.java

```diff
@@ -43,20 +48,20 @@
       EntityConstants.getValue(BackendAttribute.BACKEND_ATTRIBUTE_PORT);
 
   private BackendEntityEnricher enricher;
+  private EntityCache entityCache;
 
   @Mock
-  private EntityDataServiceClient edsClient;
+  private EdsCacheClient edsClient;
+  @Mock
+  private ClientRegistry clientRegistry;
 
   @BeforeEach
   public void setup() {
     enricher = new BackendEntityEnricher();
-    edsClient = mock(EntityDataServiceClient.class);
-    enricher.init(getEntityServiceConfig(), createProvider(edsClient));
-  }
-
-  @AfterEach
-  public void teardown() {
-    EntityCache.EntityCacheProvider.clear();
+    entityCache = new EntityCache(edsClient);
```

Comment:
Used real entitycache in the tests as they were already written to expect that (but implicitly, since it was previously retrieved internally via a static method. These tests could be improved by mocking that dependency, too.